reversed_linkedlist.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import print_function
+
+class Node(object):
+    """
+    Node implementation
+    """
+    def __init__(self, payload=None, next_node=None):
+        self.payload = payload
+        self.next_node = next_node
+
+    def get_payload(self):
+        return self.payload
+
+    def get_next(self):
+        return self.next_node
+
+    def has_next(self):
+        return bool(self.next_node)
+
+    def set_next(self, new_next):
+        self.next_node = new_next
```

Comment:
I'd say because a `Node`'s only concern is to link to the next one. Thus `set_next` makes sense.